### PR TITLE
feat(rtc): add AudioResampler, combineAudioFrames

### PR DESCRIPTION
**Pull request by nbsp**

### CODE REVIEW

File: packages/livekit-rtc/src/audio_resampler.ts [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+      return [];
+    }
+
+    const outputData = FfiClient.instance.copyBuffer(res.outputPtr, res.size);
+    return [
+      new AudioFrame(
+        new Int16Array(outputData.subarray()),
+        this.#outputRate,
+        this.#channels,
+        outputData.length,
```
Comment (theomonnom) [inline packages/livekit-rtc/src/audio_resampler.ts:124]: This doesn't seems right, samples per channel != bytes count 
You should divide by 2 and num_channels

File: packages/livekit-rtc/src/audio_resampler.ts [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
```
Comment (theomonnom) [inline packages/livekit-rtc/src/audio_resampler.ts:162]: Same here

**Review by theomonnom — CHANGES_REQUESTED** [review]

File: packages/livekit-rtc/src/audio_frame.ts [hunk 3]
```diff
@@ -52,3 +52,42 @@ export class AudioFrame {
...
+    }
+
+    if (frame.channels != channels) {
+      throw new Error(`channel mismatch: expected ${channels}, got ${frame.channels}`);
+    }
+
+    totalSamplesPerChannel += frame.samplesPerChannel;
+  }
+
+  const data = new Int16Array(buffer.map((x) => [...x.data]).flat());
```
Comment (theomonnom) [inline packages/livekit-rtc/src/audio_frame.ts:91]: Should we preallocate instead?

Comment (nbsp) [reply packages/livekit-rtc/src/audio_frame.ts:91]: i don't think node is at a level where it cares about preallocation vs conversion

File: packages/livekit-rtc/src/audio_resampler.ts (hunk 1 above)
Comment (nbsp) [reply packages/livekit-rtc/src/audio_resampler.ts:124]: looking at our own example this seems to be fine:

https://github.com/livekit/node-sdks/blob/main/examples/publish-wav/index.ts#L60

```ts
     Math.trunc(frameSize / channels),

Comment (theomonnom) [reply packages/livekit-rtc/src/audio_resampler.ts:124]: I if you check above, data size was already divided by two

**Review by bcherry — COMMENTED** [review]
this could be a good candidate for unit tests. we don't have them in the agents framework yet but it would be good to have them in the base SDK as a start. 

I added the scaffold in https://github.com/livekit/node-sdks/pull/287

**Review by theomonnom — APPROVED** [review]

**Pull request merged**